### ISSUE TITLE
I cannot get stylelint to work even though I have it configured.

### ISSUE THREAD

**Issue by Soft-Shock** [shinnn/vscode-stylelint#51]
I've installed your extension, configured it, and restarted VS Code.
VS Code: 1.9.0
OS: Windows 10

I added a config object, why is it not working? I've entered extra spaces on my indenting and added a space before a colon in a selector, no warnings are shown.

This is my user settings in VS Code (%appdata%\Code\User\settings.json). I saved and restarted VS Code after adding them.

    {    
    "stylelint.enable": true,

    "css.validate": false,

    "stylelint.config": {
        "at-rule-empty-line-before": [ "always", {
        "except": [
            "blockless-after-same-name-blockless",
            "first-nested"
        ],
        "ignore": ["after-comment"]
        } ],
        "at-rule-name-case": "lower",
        "at-rule-name-space-after": "always-single-line",
        "at-rule-semicolon-newline-after": "always",
        "block-closing-brace-empty-line-before": "never",
        "block-closing-brace-newline-after": "always",
        "block-closing-brace-newline-before": "always-multi-line",
        "block-closing-brace-space-before": "always-single-line",
        "block-no-empty": true,
        "block-opening-brace-newline-after": "always-multi-line",
        "block-opening-brace-space-after": "always-single-line",
        "block-opening-brace-space-before": "always",
        "color-hex-case": "lower",
        "color-hex-length": "short",
        "color-no-invalid-hex": true,
        "comment-empty-line-before": [ "always", {
        "except": ["first-nested"],
        "ignore": ["stylelint-commands"]
        } ],
        "comment-no-empty": true,
        "comment-whitespace-inside": "always",
        "custom-property-empty-line-before": [ "always", {
        "except": [
            "after-custom-property",
            "first-nested"
        ],
        "ignore": [
            "after-comment",
            "inside-single-line-block"
        ]
        } ],
        "declaration-bang-space-after": "never",
        "declaration-bang-space-before": "always",
        "declaration-block-no-duplicate-properties": [ true, {
        "ignore": ["consecutive-duplicates-with-different-values"]
        } ],
        "declaration-block-no-redundant-longhand-properties": true,
        "declaration-block-no-shorthand-property-overrides": true,
        "declaration-block-semicolon-newline-after": "always-multi-line",
        "declaration-block-semicolon-space-after": "always-single-line",
        "declaration-block-semicolon-space-before": "never",
        "declaration-block-single-line-max-declarations": 1,
        "declaration-block-trailing-semicolon": "always",
        "declaration-colon-newline-after": "always-multi-line",
        "declaration-colon-space-after": "always-single-line",
        "declaration-colon-space-before": "never",
        "declaration-empty-line-before": [ "always", {
        "except": [
            "after-declaration",
            "first-nested"
        ],
        "ignore": [
            "after-comment",
            "inside-single-line-block"
        ]
        } ],
        "font-family-no-duplicate-names": true,
        "function-calc-no-unspaced-operator": true,
        "function-comma-newline-after": "always-multi-line",
        "function-comma-space-after": "always-single-line",
        "function-comma-space-before": "never",
        "function-linear-gradient-no-nonstandard-direction": true,
        "function-max-empty-lines": 0,
        "function-name-case": "lower",
        "function-parentheses-newline-inside": "always-multi-line",
        "function-parentheses-space-inside": "never-single-line",
        "function-whitespace-after": "always",
        "indentation": 4,
        "keyframe-declaration-no-important": true,
        "length-zero-no-unit": true,
        "max-empty-lines": 1,
        "media-feature-colon-space-after": "always",
        "media-feature-colon-space-before": "never",
        "media-feature-name-case": "lower",
        "media-feature-name-no-unknown": true,
        "media-feature-parentheses-space-inside": "never",
        "media-feature-range-operator-space-after": "always",
        "media-feature-range-operator-space-before": "always",
        "media-query-list-comma-newline-after": "always-multi-line",
        "media-query-list-comma-space-after": "always-single-line",
        "media-query-list-comma-space-before": "never",
        "no-empty-source": true,
        "no-eol-whitespace": true,
        "no-extra-semicolons": true,
        "no-invalid-double-slash-comments": true,
        "no-missing-end-of-source-newline": true,
        "number-leading-zero": "always",
        "number-no-trailing-zeros": true,
        "property-case": "lower",
        "property-no-unknown": true,
        "rule-empty-line-before": [ "always-multi-line", {
        "except": ["first-nested"],
        "ignore": ["after-comment"]
        } ],
        "selector-attribute-brackets-space-inside": "never",
        "selector-attribute-operator-space-after": "never",
        "selector-attribute-operator-space-before": "never",
        "selector-combinator-space-after": "always",
        "selector-combinator-space-before": "always",
        "selector-descendant-combinator-no-non-space": true,
        "selector-list-comma-newline-after": "always",
        "selector-list-comma-space-before": "never",
        "selector-max-empty-lines": 0,
        "selector-pseudo-class-case": "lower",
        "selector-pseudo-class-no-unknown": true,
        "selector-pseudo-class-parentheses-space-inside": "never",
        "selector-pseudo-element-case": "lower",
        "selector-pseudo-element-colon-notation": "double",
        "selector-pseudo-element-no-unknown": true,
        "selector-type-case": "lower",
        "selector-type-no-unknown": true,
        "shorthand-property-no-redundant-values": true,
        "string-no-newline": true,
        "unit-case": "lower",
        "unit-no-unknown": true,
        "value-list-comma-newline-after": "always-multi-line",
        "value-list-comma-space-after": "always-single-line",
        "value-list-comma-space-before": "never",
        "value-list-max-empty-lines": 0
    }
    }

**Comment by shinnn** [comment]
Rules must be placed under `rules` property in `stylelint.config`. Read https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md more carefully.

**Comment by Soft-Shock** [comment]
Thank you for your help.

As a first time user of this plugin, this was very confusing to setup. Not all of us are NodeJS developers, and some of us have never even used styleLint ever, so we don't know what Object your `stylelint.config` configuration takes.

I thought the "config option" was the "rules object" for `stylelint.config` in

```
stylelint.config

Type: Object
Default: null
Will be directly passed to config option.
```

It would of been clearer to me if you said it was the Config Object for StyleLint and give a short example of the format.

Better yet,  an example of a configuration that works on the usage page would be most helpful

```
{
    "css.validate": false,

    "stylelint.enable": true,

    // Fill in your rules (or copy paste them from default config)
    "stylelint.config": {
        "rules": {
            // Fill in default Config rules from https://github.com/stylelint/stylelint-config-standard/blob/master/index.js
        }
    }
}
```

**Comment by shinnn** [comment]
Totally it's due to your carelessness.

**Comment by Soft-Shock** [comment]
Its because I'm new to this and its confusing. I didn't recklessly skip over your documentation.

Hope you're happy with the many more issues exactly like this one, because new users are not going to know how to use it.
